src/shared/components/copyDownloadControls/CopyDownloadControls.tsx

```diff
@@ -31,77 +44,154 @@ export class CopyDownloadControls extends React.Component<ICopyDownloadControlsP
+                    <If condition={this.props.showCopy}>
+                        <DefaultTooltip
+                            overlay={<span>Copy</span>}
+                            placement="top"
+                            mouseLeaveDelay={0}
+                            mouseEnterDelay={0.5}
+                            arrowContent={arrowContent}
                         >
-                            <i className='fa fa-clipboard'/>
-                        </button>
-                    </DefaultTooltip>
-                </If>
-
-                <If condition={this.props.showDownload}>
-                    <DefaultTooltip
-                        overlay={<span>Download CSV</span>}
-                        mouseLeaveDelay={0}
-                        mouseEnterDelay={0.5}
-                        placement="top"
-                        arrowContent={arrowContent}
-                    >
-                        <Button className="btn-sm" onClick={this.handleDownload}>
-                            <i className='fa fa-cloud-download'/>
-                        </Button>
-                    </DefaultTooltip>
-                </If>
-            </ButtonGroup>
+                            <button
+                                ref={(el:HTMLButtonElement) => {this._copyButton = el;}}
+                                className="btn btn-sm btn-default"
+                                data-clipboard-text="NA"
+                                id="copyButton"
+                            >
+                                <i className='fa fa-clipboard'/>
+                            </button>
+                        </DefaultTooltip>
+                    </If>
+
+                    <If condition={this.props.showDownload}>
+                        <DefaultTooltip
+                            overlay={<span>Download CSV</span>}
+                            mouseLeaveDelay={0}
+                            mouseEnterDelay={0.5}
+                            placement="top"
+                            arrowContent={arrowContent}
+                        >
+                            <Button className="btn-sm" onClick={this.handleDownload}>
+                                <i className='fa fa-cloud-download'/>
+                            </Button>
+                        </DefaultTooltip>
+                    </If>
+                </ButtonGroup>
+                <Modal
+                    show={this.downloadingData}
+                    onHide={() => undefined}
+                    bsSize="sm"
+                    className={`${copyDownloadStyles["centered-modal-dialog"]}`}
+                >
+                    <Modal.Body>
+                        <ThreeBounce style={{ display:'inline-block', marginRight:10 }} />
+                        <span>Downloading Table Data...</span>
+                    </Modal.Body>
+                </Modal>
+                <Modal
```

Comment:
We need to show another modal to be able to copy data to clipboard. It is not possible to trigger an async copy to clipboard function programmatically due to browser security restrictions.